pkg/apis/extensions/validation/validation.go

```diff
@@ -152,6 +165,8 @@ func ValidateThirdPartyResource(obj *extensions.ThirdPartyResource) field.ErrorL
 		version := &obj.Versions[ix]
 		if len(version.Name) == 0 {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("versions").Index(ix).Child("name"), version, "must not be empty"))
+		} else if !validation.IsDNS1123Label(version.Name) {
```

Comment:
I'm ok with starting tight.  But it also encourages clients to make
assumptions "oh, third party is label, I'll just mix in some colons and
underscores and everything will be safe".  No objection, otherwise LGTM.

On Mon, May 9, 2016 at 2:55 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/apis/extensions/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/25007#discussion_r62552880>
> :
>
> > @@ -152,6 +165,8 @@ func ValidateThirdPartyResource(obj *extensions.ThirdPartyResource) field.ErrorL
> >  		version := &obj.Versions[ix]
> >  		if len(version.Name) == 0 {
> >  			allErrs = append(allErrs, field.Invalid(field.NewPath("versions").Index(ix).Child("name"), version, "must not be empty"))
> > +		} else if !validation.IsDNS1123Label(version.Name) {
>
> yeah, I'm not sure what the right line to draw is. if there was a way for
> thirdpartyresources to specify what name formats they allow, that would be
> better, but in the absence of that, how much should we protect users of
> this API from bad names? see discussion at #25007 (diff)
> <https://github.com/kubernetes/kubernetes/pull/25007#discussion-diff-61665610>
>
> @liggitt <https://github.com/liggitt>: should we make the name validation
> tight (dns label or subdomain, etc) or minimal (anything we can represent
> as a URL path segment)?
>
> @deads2k <https://github.com/deads2k>: Without actual hooks to allow them
> to shape their names, I think forcing the label is reasonable.
>
> @liggitt <https://github.com/liggitt>: you don't think most API consumers
> would be equipped to handle resources named İ ♥ \n?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25007/files/6c323a4f723cbcb1a11d12edf1c81b8ed908057d#r62552880>
>